docs/how-to/create_plugin.rst

```diff
@@ -0,0 +1,131 @@
+.. _how_to_create_plugin:
+
+How to create a plugin
+======================
+
+Plugins help to bring new tools and programming languages into craft-parts.
+This document contains instructions on how to recreate the simple
+:py:class:`~craft-parts.plugins.dump_plugin.DumpPlugin` plugin.
+
+A plugin is made up of two major components: its properties and its
+functionality. The properties of a plugin are the unique pieces of
+configuration that make up the user interface of the plugin. These include
+things like specifying a Python version to use in a Python-based plugin, or
+the authentication scheme to use to connect to a remote server - it all
+depends on what your plugin does, and what it needs to do its job! The
+functionality of the plugin is the Python code that makes up its actual
+execution. The functionality of the plugin should take in configuration from
+the plugin's properties.
+
+Create your plugin file
+-----------------------
+The first step in creating a plugin is deciding where it will live. If you're
+adding a plugin directly to ``craft-parts``, then it should go in
+:file:`craft_parts/plugins`. Since we're re-creating the ``dump`` plugin,
+we'll put it in :file:`craft_parts/plugins/new_dump_plugin.py`.
+
+Imports
+~~~~~~~
+First, we'll have to pull in our imports. Since this is such a simple plugin,
+the list of necessary imports is quite small:
+
+.. code:: py
+
+    from typing import Literal
+    from overrides import override
+
+    from .base import Plugin
+    from .properties import PluginProperties
+
+Defining a properties class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+All plugins make use of a "properties" class to describe their unique parameters.
+The dump class is very basic, so we only need to define the required parameters:
```

Comment:
As a reader, I'd like to know:

- Are there _non_ unique arguments?
- Where should I go if I want non-basic arguments?

Are these relevant questions we could answer?

Reply:
- I think I'm conflating "non-unique" and "required" in a weird way here. I'll reword it. What I'm trying to say is that there are mandatory properties (such as the name of the plugin), but you can also add your own special properties (such as compiler flags or dependency files). I'll adjust.
- Answering this might go beyond scope of this how-to since we aren't using any "special properties" (as defined above), but I can add a couple of examples there or slightly modify the dump plugin to have some sort of "special property" if you think that would be useful to demonstrate. Let me know!

Reply:
The added description clarifies thing. I would call what we're describing the plugin's main interface.

I think we should dial back the amount of description in your latest commit, and compose this section like this:

> Next, define the plugin's interface. All plugins make use of a `properties` class to describe their interfaces.
There are two recommended properties that all plugins have in common, and you can add more to account for additional behaviour.
> 
> [sample code]
> 1. Add the two recommended properties:
>     -  ``plugin`` as a Pydantic literal. This property represents the exposed name of your plugin. It will be invoked in user-facing tools and configurations, such as keys that select your plugin in a Snapcraft recipe.
>     - ``source`` as a string. This property is the path to the working files of a part. It has specialised logic for different file types ranging from tarballs to Git repositories, so you can leverage it to get much broader functionality out of an otherwise simple plugin.
> 2. Add any custom properties your plugin needs. For example, a plugin based on Java might have a unique property to specify the Java version, or a Python-based plugin might have one to specify a requirements file.
> 
> The  ``new-dump`` plugin will be quite simple, so it doesn't need custom properties. 
```